### ISSUE TITLE
Interface is buggy when the top panel is showing

### ISSUE THREAD

**Issue by Aldarone** [darktable-org/darktable#10719]
**Did you buy darktable from an application store ?**

No

**Describe the bug/issue**

Darktable interface gets some bugs when the top panel is displayed : 

- Right sidebar gets pushed off the screen
- The interface freezes (found a workaround for that but its cumbersome)

**To Reproduce**

1. Hide the top banner and the top panel, at that point Darktable is behaving properly
2. Click on the top arrow to change the top banner display mode. The interface freezes.
3. Switch window and comeback on Darktable. Interface gets refreshed and become responsive again. But the right sidebar is off-screen
4. Click on the top arrow and switch windows again until the top panel is hidden. The interface is now ok again.

**Expected behavior**

The interface should not freeze and the right sidebar should not be pushed off screen.

**Screenshots**
![Screenshot-20211228-145316](https://user-images.githubusercontent.com/2682011/147575685-7418070f-4c50-4b10-bc90-83ff44eb4b07.png)
![Screenshot-20211228-145348](https://user-images.githubusercontent.com/2682011/147575695-aeff27e5-040f-4894-8f1b-06e4c41ebaf4.png)
![Screenshot-20211228-145359](https://user-images.githubusercontent.com/2682011/147575699-70413b94-0f7b-40d1-98d0-3a98250bf33a.png)

**Screencast**

https://user-images.githubusercontent.com/2682011/147575793-3a9f2271-8a31-426f-9832-2751c303f447.mp4

**Which commit introduced the error**

Noticed it after darktable 3.8 upgrade but it seems to happen with darktable 3.6 too with a blank config file.

**Platform**

```
$ darktable --version
this is darktable 3.8.0
copyright (c) 2009-2021 johannes hanika
darktable-dev@lists.darktable.org

compile options:
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support enabled, API version 8.0.0
  Colord support enabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  ImageMagick support disabled
  OpenEXR support enabled
```

* OS                : Linux - kernel 5.15.11-arch2-1
* Linux - Distro    : ArchLinux
* Memory            :  16GB
* Graphics card     : Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
* OpenCL installed  :  Yes
* OpenCL activated  : Yes
* Xorg              : XWayland 21.1.4-1
* Desktop           : SwayWM
* GTK+              : 3.24.31 & 4.4.1

**Additional context**
_Please provide any additional information you think may be useful, for example:_

 - Can you reproduce with another darktable version(s)? Yes with darktable 3.6 (didn't notice until the 3.8 upgrade though)
 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? yes
 - Do you use lua scripts? No

**Comment by jrguerreiro** [comment]
I'm seeing exactly the same behaviour when running dt 3.8 on Mac OS X 14.1.2. It didn't happen with 3.6 nor 3.7 dev. versions.

MacBook Pro Retina 13''
OS : Mac OS X 14.1.2
Processor: Intel Core i5 2,6Ghz
Memory : 8GB
Graphics card : Intel Iris 1536 MB
OpenCL installed : Yes
OpenCL activated : No

**Comment by parafin** [comment]
Can’t reproduce on macOS 10.14. Is Mac OS X 14.1.2 really OS X 10.10? Note that support for older macOS versions (before 10.14) is being deprecated and will be dropped in next major release.

As for Linux bug - I would check if it’s reproducible with Xorg and/or different wm.

**Comment by Nilvus** [comment]
I can't reproduce in Xorg/Gnome. Could you precise both what screen resolution you use?

For @Aldarone: I see you use Wayland. Just know that Wayland is not officially supported by darktable as there's actually no color management in Wayland.

**Comment by Aldarone** [comment]
My current screen resolution is 2560x1440 and I have those in my darktablerc : 

```
screen_dpi_overwrite=96
screen_ppd_overwrite=-1
```

I tried with GNOME/Xorg and I don't need to switch windows to refresh the interface when the top panel is displayed but the right sidebar is still overflowing.

Even if it works under Xorg (minus the overflowing sidebar) it's strange that this particular part of the interface triggers a rendering bug under XWayland when everything else works fine.

**Comment by Nilvus** [comment]
@Aldarone: there's no need normally now to have specific settings on those lines. Try to reset them to default settings:
```

screen_dpi_overwrite=-1
screen_ppd_overwrite=-1,000000
```

I really can't reproduce your issue. I never had overflowing, except long time ago with a really small screen (50% of yours), so your issue is really strange.

Could you reproduce your issue after changing lines above in your darktablerc file?

About Wayland, I can' answer and as I said, it's not supported by darktabe, and will not be until Wayland provide a real color management system. And it seems that this should take a long time to come... Depend on Wayland team.

**Comment by Aldarone** [comment]
I had them here because I have a 13" HiDPI screen. Switching them to the default makes the interface just a little too big for my tastes (and it doesn't solve the issues)

**Comment by jrguerreiro** [comment]
Sorry, wrong information (that's the Safari version...)! It's OS X  10.15.7

Em ter, 28 de dez de 2021 15:06, parafin ***@***.***>
escreveu:

**Comment by Aldarone** [comment]
Ha ! I just found out that if I set screen_dpi_overwrite to 50, darktable's interface is way too small to be usable but the sidebar overflowing AND the refresh issue disappears !

**Comment by Aldarone** [comment]
I was able to trigger the bug by changing the width of the sidebar.

Hiding the top panel and resize the sidebar : No issue
Showing the top panel and resize the sidebar : If the sidebar gets « too wide » it triggers the bug.

https://user-images.githubusercontent.com/2682011/147654537-9452a1f7-cb68-4caf-8ba9-3583f400b9c4.mp4

Top banner is unaffected, it's only the top panel.

Tried with XWayland on SwayWM (my standard setup) : The interface freezes
With XWayland on Gnome/Wayland : The geometry of darktable window gets weird : it gets bigger on the right side, overflowing the screen and the window cannot be maximized (option is grey when I right click on the window decoration)

I don't have time today to test with Xorg.

**Comment by dterrahe** [comment]
Is this problem reduced by #10791?

**Comment by Aldarone** [comment]
@dterrahe Is there a build somewhere I can use in order to try ?

**Comment by dterrahe** [comment]
I'm afraid you'd have to build it yourself or find someone who uses the same distro to provide one to you...

**Comment by dterrahe** [comment]
I believe this issue may have been resolved in 4.0. Please reopen if this is not the case. Many thanks!
